core/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -200,7 +201,31 @@ public MappedFieldType clone() {
         return new MappedFieldType(this);
     }
 
-    // norelease: we need to override freeze() and add safety checks that all settings are actually set
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof MappedFieldType)) return false;
```

Comment:
super.equals already ensures this, since it does: `if (getClass() != obj.getClass()) return false;`